template/go/Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM golang:1.8.3-alpine
+
+RUN apk --no-cache add curl \ 
+    && echo "Pulling watchdog binary from Github." \
+    && curl -sSL https://github.com/openfaas/faas/releases/download/0.6.5/fwatchdog > /usr/bin/fwatchdog \
+    && chmod +x /usr/bin/fwatchdog \
+    && apk del curl --no-cache
+
```

Comment:
one quick suggestion, some other pull requests that are pending are opting to run the functions as non-root users. some examples here by @ericstoekl https://github.com/openfaas/faas-cli/pull/91.

Reply:
Ha, I should know that I was one of the guilty parties behind caninothazroot.

Not even following my own advice